tests/python/sdk/test_client.py

```diff
@@ -160,6 +160,49 @@ def mocked_version(_):
     assert "Server version '0' is not compatible with SDK version" in logger_stream.getvalue()
 
 
+@pytest.mark.parametrize(
+    "server_version, supported_version",
+    [
+        ("3.2", "2"),
+        ("2", "2.1"),
+        ("2.1", "2.1"),
+        ("2.1.1", "2.1"),
+        ("2.2", "2.1"),
+        ("2.2", "2.3"),
+        ("2.1.0.dev123", "2.1.post2"),
+        ("1!1.3", "2.1"),
+        ("1!1.1.dev12", "1!1.1"),
```

Comment:
There should be some cases that test multiple supported versions. There are multiple versions listed in the actual code, after all.

Reply:
Ok, makes sense.